Bottom: dba5b611712d663d0300dc16b0dfe39cc123bfd3
Top:    89313b1ba80ad314dd3a6ef70ef1b927bfda984d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 14:16:11 +0100

Refresh of draw-c-eraserat-should-just-ga

---

diff --git a/src/draw.c b/src/draw.c
index 00ef4f7..2273d76 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1350,25 +1350,25 @@ DrawLine (LayerTypePtr Layer, LineTypePtr Line)
  * draws a ratline
  */
 void
-DrawRat (RatTypePtr Line)
+DrawRat (RatTypePtr Rat)
 {
   if (Settings.RatThickness < 20)
-    Line->Thickness = pixel_slop * Settings.RatThickness;
+    Rat->Thickness = pixel_slop * Settings.RatThickness;
   /* rats.c set VIAFLAG if this rat goes to a containing poly: draw a donut */
-  if (TEST_FLAG(VIAFLAG, Line))
+  if (TEST_FLAG(VIAFLAG, Rat))
     {
-      int w = Line->Thickness;
+      int w = Rat->Thickness;
 
       BoxType b;
 
-      b.X1 = Line->Point1.X - w * 2 - w / 2;
-      b.X2 = Line->Point1.X + w * 2 + w / 2;
-      b.Y1 = Line->Point1.Y - w * 2 - w / 2;
-      b.Y2 = Line->Point1.Y + w * 2 + w / 2;
-      AddPart(&b);
+      b.X1 = Rat->Point1.X - w * 2 - w / 2;
+      b.X2 = Rat->Point1.X + w * 2 + w / 2;
+      b.Y1 = Rat->Point1.Y - w * 2 - w / 2;
+      b.Y2 = Rat->Point1.Y + w * 2 + w / 2;
+      AddPart (&b);
     }
   else
-    AddPart (Line);
+    DrawLine (NULL, (LineType *)Rat);
 }
 
 /* ---------------------------------------------------------------------------
@@ -1513,10 +1513,10 @@ EraseRat (RatTypePtr Rat)
       b.X2 = Rat->Point1.X + w * 2 + w / 2;
       b.Y1 = Rat->Point1.Y - w * 2 - w / 2;
       b.Y2 = Rat->Point1.Y + w * 2 + w / 2;
-      AddPart(&b);
+      AddPart (&b);
     }
   else
-    AddPart (Rat);
+    EraseLine ((LineType *)Rat);
 }
